set.mm

```diff
@@ -472429,6 +472712,198 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
+      BHTEUCZUDVNVCVJDVDVKVOVNVBVIUSVNVAVHUTVNUPAURVPUEUFULUAUGUHCDFUIVMCVKBOUJ
+      UKUM $.
+
+    $d B i $.  $d M i j $.  $d R i j $.
+    $( The predicate "is a maximal ideal".  (Contributed by Jeff Madsen,
+       5-Jan-2011.)  (Revised by Thierry Arnoux, 19-Jan-2024.) $)
+    ismxidl $p |- ( R e. Ring -> ( M e. ( MaxIdeal ` R ) <-> ( M e. ( LIdeal `
+       R ) /\ M =/= B /\ A. j e. ( LIdeal ` R ) ( M C_ j -> ( j = M \/ j = B )
+       ) ) ) ) $=
+      ( vi crg wcel cmxidl cfv cv wne wss wceq wo wi clidl wral wa crab imbi12d
+      w3a mxidlval eleq2d neeq1 sseq1 eqeq2 orbi1d ralbidv anbi12d elrab 3anass
+      bitr4i syl6bb ) BGHZDBIJZHDFKZALZUQCKZMZUSUQNZUSANZOZPZCBQJZRZSZFVETZHZDV
+      EHZDALZDUSMZUSDNZVBOZPZCVERZUBZUOUPVHDABFCEUCUDVIVJVKVPSZSVQVGVRFDVEUQDNZ
+      URVKVFVPUQDAUEVSVDVOCVEVSUTVLVCVNUQDUSUFVSVAVMVBUQDUSUGUHUAUIUJUKVJVKVPUL
+      UMUN $.
+
+    $( A maximal ideal is an ideal.  (Contributed by Jeff Madsen, 5-Jan-2011.)
+       (Revised by Thierry Arnoux, 19-Jan-2024.) $)
+    mxidlidl $p |- ( ( R e. Ring /\ M e. ( MaxIdeal ` R ) ) -> M e. ( LIdeal `
+       R ) ) $=
+      ( vj crg wcel cmxidl cfv wa clidl wne cv wss wceq wo wi wral w3a ismxidl
+      biimpa simp1d ) BFGZCBHIGZJCBKIZGZCALZCEMZNUHCOUHAOPQEUERZUCUDUFUGUISABEC
+      DTUAUB $.
+
+    $( A maximal ideal is proper.  (Contributed by Jeff Madsen, 16-Jun-2011.)
+       (Revised by Thierry Arnoux, 19-Jan-2024.) $)
+    mxidlnr $p |- ( ( R e. Ring /\ M e. ( MaxIdeal ` R ) ) -> M =/= B ) $=
+      ( vj crg wcel cmxidl cfv wa clidl wne cv wss wceq wo wi wral w3a ismxidl
+      biimpa simp2d ) BFGZCBHIGZJCBKIZGZCALZCEMZNUHCOUHAOPQEUERZUCUDUFUGUISABEC
+      DTUAUB $.
+
+    $d B j $.  $d I j $.
+    $( A maximal ideal is a maximal proper ideal.  (Contributed by Jeff Madsen,
+       16-Jun-2011.)  (Revised by Thierry Arnoux, 19-Jan-2024.) $)
+    mxidlmax $p |- ( ( ( R e. Ring /\ M e. ( MaxIdeal ` R ) )
+         /\ ( I e. ( LIdeal ` R ) /\ M C_ I ) ) -> ( I = M \/ I = B ) ) $=
+      ( vj crg wcel cmxidl cfv wa clidl wss wceq wo cv wi sseq2 eqeq1 orbi12d
+      imbi12d wral wne w3a ismxidl biimpa simp3d adantr simpr rspcdva impr ) BG
+      HZDBIJHZKZCBLJZHZDCMZCDNZCANZOZUNUPKDFPZMZVADNZVAANZOZQZUQUTQFUOCVACNZVBU
+      QVEUTVACDRVGVCURVDUSVACDSVACASTUAUNVFFUOUBZUPUNDUOHZDAUCZVHULUMVIVJVHUDAB
+      FDEUEUFUGUHUNUPUIUJUK $.
+
+    ${
+      mxidln1.1 $e |- .1. = ( 1r ` R ) $.
+      $( One is not contained in any maximal ideal.  (Contributed by Jeff
+         Madsen, 17-Jun-2011.)  (Revised by Thierry Arnoux, 19-Jan-2024.) $)
+      mxidln1 $p |- ( ( R e. Ring /\ M e. ( MaxIdeal ` R ) ) -> -. .1. e. M )
+      $=
+        ( crg wcel cmxidl cfv wa wn mxidlnr clidl wceq wb mxidlidl eqid lidl1el
+        wne syldan necon3bbid mpbird ) BGHZDBIJHZKZCDHZLDATABDEMUFUGDAUDUEDBNJZ
+        HUGDAOPABDEQABUHCDUHREFSUAUBUC $.
+    $}
+
+    $( A ring with a maximal ideal is a nonzero ring.  (Contributed by Jeff
```

Comment:
If I recall correctly, every nonzero ring (with 1) has a maximal ideal, so maybe it's worth strengthening this theorem.

Reply:
Gérard also suggested to prove a similar statement, that for any proper ideal $I$ of a ring $R$, there is a maximal ideal of $R$ containing $I$.
I'll try to include that in a subsequent PR (using Zorn's Lemma), and it should then be possible to derive [Krull's theorem](https://en.wikipedia.org/wiki/Krull%27s_theorem), which is what you stated.

Reply:
See #3912 for a follow-up on this.